modules/client/src/internal/client/decoding.ts

```diff
@@ -125,7 +129,7 @@ export class ClientDecoding extends ClientCore implements IClientDecoding {
     }
     const result = daoInterface.decodeFunctionData("setMetadata", data);
     const bytes = hexToBytes(result[0]);
-    const cid = new TextDecoder().decode(bytes);
+    const cid = new TextDecoder().decode(bytes).replace("ipfs://", "");
```

Comment:
At least temporarily, but I'd make sure that the value contains `ipfs://<cid>`. Otherwise I would throw an error.

We may add further URI types support, but until then I would validate at least this.